crates/ethereum/evm/src/lib.rs

```diff
@@ -255,10 +239,15 @@ impl ConfigureEvm for EthEvmConfig {
     type Evm<'a, DB: Database + 'a, I: 'a> = EthEvm<'a, I, DB>;
 
     fn evm_with_env<DB: Database>(&self, db: DB, evm_env: EvmEnv) -> Self::Evm<'_, DB, ()> {
+        let cfg_env_with_handler_cfg = CfgEnvWithHandlerCfg {
+            cfg_env: evm_env.cfg_env,
+            #[allow(clippy::needless_update)] // side-effect of optimism fields
+            handler_cfg: HandlerCfg { spec_id: evm_env.spec, ..Default::default() },
```

Comment:
can we use `HandlerCfg::new` here?
iirc this has an internal cfg for the optimism flag

Reply:
I guess we'll need to change this soon anayway so we could keep just to ensure we're setting optimism to false in case the feature is enabled outside of reth for whatever reason